src/useMintUsd.ts

```diff
@@ -0,0 +1,105 @@
+import { useMutation } from '@tanstack/react-query';
+import type { ethers } from 'ethers';
+import { fetchMintUsd } from './fetchMintUsd';
+import { fetchMintUsdWithPriceUpdate } from './fetchMintUsdWithPriceUpdate';
+import { fetchPriceUpdateTxn } from './fetchPriceUpdateTxn';
+import { useImportContract } from './useImports';
+import { useSynthetix } from './useSynthetix';
+
+interface CollateralType {
+  address: string;
+  decimals: number;
+  depositingEnabled: boolean;
+  issuanceRatioD18: ethers.BigNumber;
+  liquidationRatioD18: ethers.BigNumber;
+  liquidationRewardD18: ethers.BigNumber;
+  minDelegationD18: ethers.BigNumber;
+  name: string;
+  oracleNodeId: string;
+  symbol: string;
+  tokenAddress: string;
+}
+
+export function useMintUsd({
+  provider,
+  priceIds,
+  walletAddress,
+  accountId,
+  collateralType,
+  poolId,
+  onSuccess,
+}: {
+  provider?: ethers.providers.Web3Provider;
+  priceIds?: string[];
+  walletAddress?: string;
+  accountId?: ethers.BigNumber;
+  collateralType?: CollateralType;
```

Comment:
I dont think we need whole collateral type here at all.
Just a subset of things we care about

```suggestion
  collateralType?: {
    address: string;
  };
```

Let's not introduce any special Types as much as possible and rely on simple types please

Reply:
fix, add just `collateralTokenAddress`